### ISSUE TITLE
Expose maxIteration to the user when performing computeInitialFiberEquilibrium

### ISSUE THREAD

**Issue by mitkof6** [opensim-org/opensim-core#2585]
I was performing MuscleAnalysis on the gait2392/54 models (with the kinematics that come with v3.3) to calculate the fiber length and velocity of the muscles. It is important to equilibrate the state of the muscles before the analysis. However, when inspecting the length and velocities of some muscles (e.g., vas_int or soleus) there appear to be spikes and discontinuities. The tool correctly warned that the computeInitialFiberEquilibrium was not able to converge to a solution. To overcome this issue I had to disable tendon compliance so that I will get smooth results. However, the problem can be solved if the maximum iterations parameter is exposed to the user.

**Comment by mitkof6** [comment]
I also tested this with different muscle models (Millard, Thelen) and on OpenSim 3.3/4.0.

**Comment by jimmyDunne** [comment]
Thanks, @mitkof6! It looks like this is currently hardcoded, for example in the [Thelen2003Muscle](https://github.com/opensim-org/opensim-core/blob/2fc6dc1d23734f3b5f8d58c3b3833871e756c675/OpenSim/Actuators/Thelen2003Muscle.cpp#L342). It would be good to have this as changeable by the user. Though, I am struggling to think about how that would be implemented. Would you have a  a single method on a model that would update the maxIter for every muscle or would you have to change this for each muscle, individually? Do you have ideas for this?

**Comment by tkuchida** [comment]
Might be simpler to permit more iterations (up to some fixed higher limit) if the algorithm is making progress (which would need to be monitored).

**Comment by jimmyDunne** [comment]
Good point, @tkuchida. It would be good if there was some internal evaluation on progress with some high upper bound just in case. That may take some more development time but would be worth it. I don't think this will make it into the next Update but we will keep on the development list for upcoming sprints.
